docs/source-control/git-guidance/README.md

```diff
@@ -232,6 +232,74 @@ Then you can use the reflog reference (`HEAD@{}`) to reset to a specific commit
 git reset HEAD@{2}
 ```
 
+## Commit Best Practices
+
+* Make small commits. This makes changes easier to review, and if we need to revert a commit, we lose less work.
+* Commit complete and well tested code. Never commit incomplete code, get in the habit of testing your code before committing.
+* Don't mix whitespace changes with functional code changes. It is hard to determine if the line has a functional change or only removes a whitespace, so functional changes may go unnoticed.
+* Write good commit messages.
+
+## Commit Messages
+
+A commit message describes the changes made in the commit. A descriptive commit message can help understanding the code changes, and to understand why a change was made.
+A good commit message should answer these questions:
+
+* Why is it necessary? It may fix a bug, add a feature, improve performance, or just be a change for the sake of correctness
+* How does it address the issue? For short, obvious changes, this can be omitted
+* What effects does this change have? In addition to the obvious ones, this may include benchmarks, side effects etc.
+* What limitations does the current code have?
+
+Consider this when writing your commit message:
+
+* Don't assume that the code is self-evident/self-documenting
+* If it seems difficult to summarize your commit, it may be because it includes more than one logical change or bug fix. If so, it is better to split it into separate commits with `git add -p`
+* Don't assume the reviewer understands the original problem. It should be possible to review a change request without reading the contents of the original bug/task.
+
+You can specify the default git editor, which allows you to write your commit messages using your favorite editor. The following command makes Visual Studio Code your default git editor:
+
+```bash
+git config --global core.editor "code --wait"
+```
+
+### Commit Message Structure
+
+You are free to structure commit messages. However, it can be helpful to follow a convention within your team.
```

Comment:
I think all this content comes from a post on "how to write good commits" - maybe we can simply remove all of this and link to the post itself and thus make the documentation here a lot cleaner.